app/models/slug_migration.rb

```diff
@@ -0,0 +1,30 @@
+class SlugMigration < ActiveRecord::Base
+  belongs_to :guide
+
+  validates :slug, uniqueness: true
+  validate :guide_must_be_nil_or_published
+  validate :guide_cant_be_empty_when_migrating
+  validate :is_not_already_completed
+
+  before_validation on: :create do |object|
+    object.content_id = SecureRandom.uuid
+  end
+
```

Comment:
Minor: I like putting validation methods under `private` to reduce the public class API as they should never be called from outside the class.